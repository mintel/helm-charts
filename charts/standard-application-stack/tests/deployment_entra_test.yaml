suite: Test Entra env-vars injected into Deployment
templates:
  - deployment.yaml
release:
  namespace: test-namespace
tests:
  - it: Check entra client-secrets are added
    set:
      global.clusterEnv: qa
      global.name: test-app
      entra.enabled: true
      entra.includeClientSecretsInWorkload: true
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: Deployment
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: spec.template.spec.containers[0].env
          value:
            - name: KUBERNETES_POD_NAMESPACE
              valueFrom:
                fieldRef:
                  fieldPath: metadata.namespace
            - name: APP_ENVIRONMENT
              value: qa
            - name: RUNTIME_ENVIRONMENT
              value: kubernetes
            - name: AZURE_CLIENT_ID
              valueFrom:
                secretKeyRef:
                  key: clientID
                  name: test-app-ingress-oidc-credentials
            - name: AZURE_CLIENT_SECRET
              valueFrom:
                secretKeyRef:
                  key: clientSecret
                  name: test-app-ingress-oidc-credentials
  - it: Check entra client-secrets are excluded when not enabled
    set:
      global.clusterEnv: qa
      global.name: test-app
      entra.enabled: true
      entra.includeClientSecretsInWorkload: false
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: Deployment
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: spec.template.spec.containers[0].env
          value:
            - name: KUBERNETES_POD_NAMESPACE
              valueFrom:
                fieldRef:
                  fieldPath: metadata.namespace
            - name: APP_ENVIRONMENT
              value: qa
            - name: RUNTIME_ENVIRONMENT
              value: kubernetes
