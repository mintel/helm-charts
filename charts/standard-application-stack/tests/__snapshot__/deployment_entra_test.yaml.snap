Check entra client-secrets are added:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        secret.reloader.stakater.com/reload: test-app-app
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/application: test-app
        app.mintel.com/component: test-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      minReadySeconds: 10
      replicas: 2
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
      strategy:
        rollingUpdate:
          maxSurge: 15%
          maxUnavailable: 10%
        type: RollingUpdate
      template:
        metadata:
          labels:
            app.kubernetes.io/component: app
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: test-app
            app.mintel.com/application: test-app
            app.mintel.com/component: test-app
            app.mintel.com/env: qa
            app.mintel.com/region: ${CLUSTER_REGION}
            name: test-app
        spec:
          containers:
            - command:
                - /app/docker-entrypoint.sh
              env:
                - name: KUBERNETES_POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: APP_ENVIRONMENT
                  value: qa
                - name: RUNTIME_ENVIRONMENT
                  value: kubernetes
                - name: AZURE_CLIENT_ID
                  valueFrom:
                    secretKeyRef:
                      key: clientId
                      name: test-app-ingress-oidc-credentials
                - name: AZURE_CLIENT_SECRET
                  valueFrom:
                    secretKeyRef:
                      key: clientSecret
                      name: test-app-ingress-oidc-credentials
              envFrom:
                - secretRef:
                    name: test-app-app
              image: registry.gitlab.com/test:v0.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 10
                timeoutSeconds: 1
              name: main
              ports:
                - containerPort: 8000
                  name: http
              readinessProbe:
                httpGet:
                  path: /readiness
                  port: http
                  scheme: HTTP
                periodSeconds: 10
                timeoutSeconds: 1
              resources:
                limits: {}
                requests: {}
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                runAsNonRoot: true
                runAsUser: 1000
              startupProbe:
                failureThreshold: 60
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 5
          securityContext:
            runAsNonRoot: true
            runAsUser: 1000
          serviceAccountName: test-app
          terminationGracePeriodSeconds: 30
          topologySpreadConstraints:
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/component: app
                  app.kubernetes.io/name: test-app
              maxSkew: 1
              topologyKey: topology.kubernetes.io/zone
              whenUnsatisfiable: DoNotSchedule
Check entra client-secrets are excluded when not enabled:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        secret.reloader.stakater.com/reload: test-app-app
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/application: test-app
        app.mintel.com/component: test-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      minReadySeconds: 10
      replicas: 2
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
      strategy:
        rollingUpdate:
          maxSurge: 15%
          maxUnavailable: 10%
        type: RollingUpdate
      template:
        metadata:
          labels:
            app.kubernetes.io/component: app
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: test-app
            app.mintel.com/application: test-app
            app.mintel.com/component: test-app
            app.mintel.com/env: qa
            app.mintel.com/region: ${CLUSTER_REGION}
            name: test-app
        spec:
          containers:
            - command:
                - /app/docker-entrypoint.sh
              env:
                - name: KUBERNETES_POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: APP_ENVIRONMENT
                  value: qa
                - name: RUNTIME_ENVIRONMENT
                  value: kubernetes
              envFrom:
                - secretRef:
                    name: test-app-app
              image: registry.gitlab.com/test:v0.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 10
                timeoutSeconds: 1
              name: main
              ports:
                - containerPort: 8000
                  name: http
              readinessProbe:
                httpGet:
                  path: /readiness
                  port: http
                  scheme: HTTP
                periodSeconds: 10
                timeoutSeconds: 1
              resources:
                limits: {}
                requests: {}
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                runAsNonRoot: true
                runAsUser: 1000
              startupProbe:
                failureThreshold: 60
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 5
          securityContext:
            runAsNonRoot: true
            runAsUser: 1000
          serviceAccountName: test-app
          terminationGracePeriodSeconds: 30
          topologySpreadConstraints:
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/component: app
                  app.kubernetes.io/name: test-app
              maxSkew: 1
              topologyKey: topology.kubernetes.io/zone
              whenUnsatisfiable: DoNotSchedule
